### PR TITLE
Add basic user home a/a experiment

### PR DESCRIPTION
**Pull request by withinboredom**
#### Changes proposed in this Pull Request

We are looking to do an alpha test of the experimentation framework. This PR temporarily adds a simple A/A test to user home. It does not change the structure or content of the page.

#### Testing instructions

1. Visit user home
2. Visit experiments-dev for `user_home_test`
3. See that you're assigned to the default
4. Assign yourself to another variation, see no changes on user home.

pbmo2S-lg-p2

### CODE REVIEW

**Comment by gwwar** [comment]
For folks reading, note that depending on what the test is a server side test would likely be more appropriate for the My Home section.

**Comment by gwwar** [comment]
@withinboredom are we waiting on any other changes, or is this one ready for review? How long are we planning on leaving the A/A test in?

**Comment by withinboredom** [comment]
@gwwar I'm not anticipating this experiment being in this code longer than 3 weeks:
1. Week one will be enabling the feature flag in environments and looking for unexpected failures as we go.
2. Week two will be actually enabling the feature and making sure analysis collects data correctly, along with audits of the frameworks.
3. Week three, the experiment will be completed and automatically transition to the winning state, so we'll audit that as well.
4. We'll revert (or remove) the experiment.

> is this one ready for review?

Yep, I'd love a review of this!

File: client/my-sites/customer-home/main.jsx [hunk 1]
```diff
@@ -49,6 +50,14 @@ const Home = ( {
 	trackViewSiteAction,
 } ) => {
 	const translate = useTranslate();
+	// use an effect to fire the exposure event only if the site id changes or first view
```
Comment (gwwar) [inline client/my-sites/customer-home/main.jsx:53]: Maybe leave a comment here that we're temporarily testing the experimentation framework as an A/A test and that we expect to remove this in +3 weeks or so.

File: client/my-sites/customer-home/main.jsx [hunk 2]
```diff
@@ -49,6 +50,14 @@ const Home = ( {
...
+	useEffect( () => {
+		if ( canUserUseCustomerHome ) {
+			recordTracksEvent( 'calypso_saw_a_a_test', {
```
Comment (gwwar) [inline client/my-sites/customer-home/main.jsx:56]: This needs to be connected in mapDispatchToProps, as I don't see this being fired.

To debug analytic events try setting in dev console: `localStorage.debug = 'calypso:analytics*'` and refresh the page.

We should see events like the following in console:
Screen Shot 2020-05-12 at 2 39 05 PM

**Comment by gwwar** [comment]
cc @scinos would landing this affect any upcoming RUM experimentation? I know y'all are planning to use home as one of the first sections to monitor.

**Comment by scinos** [comment]
@gwwar Thanks for asking. It shouldn't affect for the RUM experiments.

But for the final RUM we'll probably add a flag with the active experiments, so we can segment perf data based on them. I'll ping you when we are ready to add those flags.

**Review by mmtr — COMMENTED** [review]
Overall this tests well for me. I see the same header in both variations, but I verified it is rendered by the right component.

`Variation` renders the header when I'm assigned to `treatment`.
Screen Shot 2020-05-13 at 11 55 09

`DefaultVariation ` renders the header when I'm assigned to `control`.
Screen Shot 2020-05-13 at 11 55 34

Events are not tracked because we're not dispatching the action correctly, but other than that this works great. I also left a non-blocking note about moving the event tracking to the `Experiment` component, so it's even easier to create experiments.

File: client/my-sites/customer-home/main.jsx (hunk 2 above)
Comment (withinboredom) [reply client/my-sites/customer-home/main.jsx:56]: Good catch! I was wondering why I wasn't seeing that event but figured it was me or my environment.

[the remark this answers is not in the document]
Comment (withinboredom) [reply client/my-sites/customer-home/main.jsx]: As soon as I realized the complexity for adding that prop, I just decided to implement it on the experiment component. This way I could add some tests too.

File: client/my-sites/customer-home/main.jsx (hunk 1 above)
Comment (withinboredom) [reply client/my-sites/customer-home/main.jsx]: I moved this to the experiment component, so as to not bloat this component (and future experiments)

File: client/my-sites/customer-home/main.jsx [hunk 3]
```diff
@@ -60,27 +60,38 @@ const Home = ( {
...
-					) }
-					align="left"
-				/>
-				<div className="customer-home__view-site-button">
-					<Button href={ site.URL } onClick={ trackViewSiteAction }>
-						{ translate( 'View site' ) }
-					</Button>
-				</div>
-			</div>
+			<Experiment name="user_home_test" event={ 'calypso_saw_a_a_test' }>
```
Comment (mmtr) [inline client/my-sites/customer-home/main.jsx:85]: Nitpick:
```suggestion
			<Experiment name="user_home_test" event="calypso_saw_a_a_test">
```

File: client/my-sites/customer-home/main.jsx [hunk 4]
```diff
@@ -60,27 +60,38 @@ const Home = ( {
...
+				<LoadingVariations>
+					{ /* Normally, we'd load variations before this page could be visible. However, we'll just pretend
+					     we did so that users have an unchanged experience.
+					   */ }
+					{ header }
+				</LoadingVariations>
+				<DefaultVariation name={ 'control' }>{ header }</DefaultVariation>
```
Comment (mmtr) [inline client/my-sites/customer-home/main.jsx:92]:
```suggestion
				<DefaultVariation name="control">{ header }</DefaultVariation>
```

File: client/my-sites/customer-home/main.jsx [hunk 5]
```diff
@@ -60,27 +60,38 @@ const Home = ( {
...
+				<Variation name={ 'treatment' }>{ header }</Variation>
```
Comment (mmtr) [inline client/my-sites/customer-home/main.jsx:93]:
```suggestion
				<Variation name="treatment">{ header }</Variation>
```

**Review by mmtr — COMMENTED** [review]
Thanks @withinboredom for abstracting the event tracking. I can see them when debugging the analytics.

Treatment:
Screen Shot 2020-05-13 at 16 53 07

Control:
Screen Shot 2020-05-13 at 16 52 37

Question: Is the `null` variation the expected value when tracking the event for users assigned to the control variation? Or should it be `control`?

**Comment by withinboredom** [comment]
Yes, `null` is expected for the default. We decided to make it so application code can’t distinguish between when a user is not in an experiment and when a user is just assigned to the control. This way we knew the “control” was really a control and not another variation. There was also some back and forth about whether to have the `name` prop for the default since it isn’t necessary. I’m interested to hear another persons thoughts now that’s it’s actually implemented (but probably outside the scope of this pr).

**Comment by mmtr** [comment]
Thanks for clarifying.

> There was also some back and forth about whether to have the `name` prop for the default since it isn’t necessary.

Yeah, I think it'd be nice if a `name` prop is not required for the default variation, but agree that is beyond the scope of this PR.

**Review by mmtr — APPROVED** [review]

File: client/my-sites/customer-home/main.jsx (hunk 3 above)
Comment (aaronyan) [inline client/my-sites/customer-home/main.jsx:85]: Consider changing the event name to be `calypso_aatest_start` to follow the naming pattern of the existing `calypso_abtest_start` event.
```suggestion
			<Experiment name="user_home_test" event="calypso_aatest_start">>
```

Comment (withinboredom) [reply client/my-sites/customer-home/main.jsx:85]: I'd rather not anyone ever use this event again.

Comment (withinboredom) [reply client/my-sites/customer-home/main.jsx:85]: But in all seriousness, it has very different semantics than the abtest_start event.

Comment (withinboredom) [reply client/my-sites/customer-home/main.jsx]: applied in 8faae87

File: client/my-sites/customer-home/main.jsx (hunk 4 above)
Comment (withinboredom) [reply client/my-sites/customer-home/main.jsx]: applied in 8faae87

File: client/my-sites/customer-home/main.jsx (hunk 5 above)
Comment (withinboredom) [reply client/my-sites/customer-home/main.jsx]: applied in 8faae87

File: client/my-sites/customer-home/main.jsx (hunk 3 above)
Comment (aaronyan) [reply client/my-sites/customer-home/main.jsx]: That's fair. Yeah, if we change the name, we should go all the way and also match the eventprops like `abtest_name` and `abtest_variation`.

Comment (withinboredom) [reply client/my-sites/customer-home/main.jsx]: It's also just to make things easier for cases where there's already not an exposure event readily available. AFAIK, the event props isn't used so much (unless specified in the experiment definition). In fact, an event is not required at all, I just did it so we could enable all the analysis types and see how they compare. If anything, I suggest making the event name more "ridiculous" just to make it obvious that the event name doesn't matter.

Comment (yanirs) [reply client/my-sites/customer-home/main.jsx]: Agreed with not using `calypso_aatest_start`, but `calypso_saw_a_a_test` doesn't match [the Tracks naming conventions](https://fieldguide.automattic.com/tracks/tracks-for-developers-product-teams/tracks-naming-conventions/). Let's go with something more compliant, like `calypso_customer_home_aa_experiment_exposed`.

**Review by yanirs — CHANGES_REQUESTED** [review]
I don't understand how exposure events crept into this PR after all our discussions on it *not* being the responsibility of the client! 

There are multiple problems with this approach:

* It encourages bad experimentation practices and event inflation. The idea behind exposure events is to use _existing_ events and/or keep event firing decoupled from the experimentation framework, not to get a new event for every experiment!
* We don't need a new event to track exposure in this case. We have `calypso_page_view` with `path = /home/:site`. 
* With the current implementation, `calypso_saw_a_a_test` fired three times when I loaded the customer home.

Please undo all the changes under `/client/components/experiment` and use the client as we originally discussed. :slightly_smiling_face:

File: client/my-sites/customer-home/main.jsx (hunk 3 above)
Comment (yanirs) [reply client/my-sites/customer-home/main.jsx]: Scrap that. We shouldn't be firing an event at all: https://github.com/Automattic/wp-calypso/pull/42093#pullrequestreview-411473293. It isn't the responsibility of the experimentation client and it encourages bad practices and event inflation.

**Comment by withinboredom** [comment]
I've remove the event code (smart idea to use `calypso_page_view`!) and reconfigured the experiment.

**Review by yanirs — APPROVED** [review]
Nice and simple! :ship:

**Pull request merged**